sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/GaugeData.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.metrics;
+
+import com.google.auto.value.AutoValue;
+
+import java.io.Serializable;
+
+/**
+ * Data describing the the gauge. This should retain enough detail that it can be combined with
+ * other {@link GaugeData}.
+ *
+ * <p>This is kept distinct from {@link GaugeResult} to allow generifcation of the metrics reporting
+ * process in the future.
+ */
+@AutoValue
+public abstract class GaugeData implements Serializable {
+
+  public static final GaugeData EMPTY = create(-1L);
+
+  public abstract long value();
+
+  public static GaugeData create(long value) {
+    return new AutoValue_GaugeData(value);
+  }
+
+  public GaugeData combine(GaugeData other) {
```

Comment:
It might also be worth having a "CombineMany" that is takes `Iterable<GaugeData>` and returns `GaugeData` without needing to create all the intermediate pairings.